### PR TITLE
feat(blade): Counter design changes

### PR DESCRIPTION
**Pull request by anuraghazra**
In This PR: 

1. Detached counter from badge - copy pasted logic from badge
2. Fixed counter's padding
2. Fixed counter's text size for large 

TODO: 

- Incorporate the changes which chaitanya is making in badge 
   - Adding new italic font
   - Adding _color support for Text component
   
   
   
 Web: 
 
 
image

ReactNative: 

image

### CODE REVIEW

File: packages/blade/src/components/Typography/Text/Text.tsx [hunk 1]
```diff
@@ -12,19 +12,19 @@ type TextCommonProps = {
   contrast?: ColorContrastTypes;
   truncateAfterLines?: number;
   children: React.ReactNode;
+  weight?: keyof Theme['typography']['fonts']['weight'];
+  _color?: BaseTextProps['color'];
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Text/Text.tsx:16]: TODO: Remove these changes if #837 merges before this.

**Comment by anuraghazra** [comment]
TODO: 

Italic font will be added by #837 
https://github.com/razorpay/blade/pull/837/commits/e4e092937dd14ec5da8dcafad3f8cfa2c3ad23a8

File: packages/blade/src/components/Typography/Text/Text.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Typography/Text/Text.tsx:16]: Not needed anymore since this PR will be merged first.

Comment (divyanshu013) [reply packages/blade/src/components/Typography/Text/Text.tsx:16]: 🥇

File: packages/blade/src/components/Counter/counterTokens.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import type { CounterProps } from './Counter';
+import type { TypographyPlatforms } from '~tokens/global/typography';
+import type { DotNotationSpacingStringToken } from '~src/_helpers/types';
+
+type BadgeMaxWidth = 100 | 120;
```
Comment (divyanshu013) [inline packages/blade/src/components/Counter/counterTokens.ts:5]: You probably mean:

```suggestion
type CounterMaxWidth = 100 | 120;
```

File: packages/blade/src/components/Counter/counterTokens.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+const horizontalPadding: Record<
+  NonNullable<CounterProps['size']>,
+  DotNotationSpacingStringToken
+> = {
+  small: 'spacing.2',
+  medium: 'spacing.3',
+  large: 'spacing.3',
+};
+
+const maxWidth: Record<NonNullable<TypographyPlatforms>, BadgeMaxWidth> = {
```
Comment (divyanshu013) [inline packages/blade/src/components/Counter/counterTokens.ts:22]:
```suggestion
const maxWidth: Record<NonNullable<TypographyPlatforms>, CounterMaxWidth> = {
```

**Review by divyanshu013 — COMMENTED** [review]
Looks good overall. Minor comments and pending conclusion on 
- [ ] Add changelog

File: packages/blade/src/components/Counter/counterTokens.ts (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/src/components/Counter/counterTokens.ts:5]: haha yeah

File: packages/blade/src/components/Typography/Text/Text.tsx [hunk 4]
```diff
@@ -13,7 +13,13 @@ type TextCommonProps = {
   truncateAfterLines?: number;
   children: React.ReactNode;
   weight?: keyof Theme['typography']['fonts']['weight'];
-  _color?: BaseTextProps['color'];
+  /**
+   * Sets the color of the Text component
+   *
+   * ## Only for internal use
+   *
+   */
```
Comment (divyanshu013) [inline packages/blade/src/components/Typography/Text/Text.tsx:21]: Maybe we can make it more obvious:

```suggestion
  /**
   * **Internal only**: Sets the color of the Text component
   *
   */
```

Comment (anuraghazra) [reply packages/blade/src/components/Typography/Text/Text.tsx:21]: Okay sure.

File: packages/blade/src/components/Counter/counterTokens.ts (hunk 2 above)
Comment (nashcheez) [reply packages/blade/src/components/Counter/counterTokens.ts]: @anuraghazra You suck at doing a copy, paste. I can imagine how cheating in an exam would be risky with you! ;)

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: .changeset/cyan-ducks-chew.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+"@razorpay/blade": minor
+---
+
+feat(blade): Counter design changes
```
Comment (divyanshu013) [inline .changeset/cyan-ducks-chew.md:5]:
```suggestion
feat(Counter): design changes
```

**Pull request merged**